service/resource/deployment/resource.go

```diff
@@ -181,27 +181,29 @@ func (r *Resource) GetCreateState(ctx context.Context, obj, currentState, desire
 		}
 	}
 
-	return deploymentsToCreate, nil
-}
+	patch := framework.NewPatch()
+
+	if len(deploymentsToCreate) > 0 {
+		patch.SetCreateChange(deploymentsToCreate)
+	}
```

Comment:
I would like to keep it the way we did it so far. There are `New*Patch` methods which are all the same and only call `new*Change` methods. Keeping this consistent makes it easier later on to refactor and unify. The tests can then stay as they are and test the private methods since there is no real business logic in the `New*Patch` methods.

Reply:
Done.